Mathlib/Analysis/NormedSpace/PiTensorProduct/ProjectiveSeminorm.lean

```diff
@@ -0,0 +1,155 @@
+  simp only [projectiveSeminormAux, Function.comp_apply, Multiset.map_coe, Multiset.sum_coe]
+  erw [List.map_append]
+  rw [List.sum_append]
+  rfl
+
+theorem projectiveSeminormAux_smul (p : FreeAddMonoid (𝕜 × Π i, E i)) (a : 𝕜) :
+    projectiveSeminormAux (List.map (fun (y : 𝕜 × Π i, E i) ↦ (a * y.1, y.2)) p) =
+    ‖a‖ * projectiveSeminormAux p := by
+  simp only [projectiveSeminormAux, Function.comp_apply, Multiset.map_coe, List.map_map,
+    Multiset.sum_coe]
+  rw [← smul_eq_mul, List.smul_sum, ← List.comp_map]
+  congr 2
+  ext x
+  simp only [Function.comp_apply, norm_mul, smul_eq_mul]
+  rw [mul_assoc]
+
+theorem projectiveSemiNormAuxBddBelow (x : ⨂[𝕜] i, E i) :
+    BddBelow (Set.range (fun (p : lifts x) ↦ projectiveSeminormAux p.1)) := by
+  existsi 0
+  rw [mem_lowerBounds]
+  simp only [Set.mem_range, Subtype.exists, exists_prop, forall_exists_index, and_imp,
+    forall_apply_eq_imp_iff₂]
+  exact fun p _ ↦ projectiveSeminormAux_nonneg p
+
+/-- The projective seminorm on `⨂[𝕜] i, Eᵢ`.
+-/
+noncomputable def projectiveSeminorm : Seminorm 𝕜 (⨂[𝕜] i, E i) := by
+  refine Seminorm.ofSMulLE (fun x ↦ iInf (fun (p : lifts x) ↦ projectiveSeminormAux p.1)) ?_ ?_ ?_
+  · refine le_antisymm ?_ ?_
+    · refine ciInf_le_of_le (projectiveSemiNormAuxBddBelow (0 : ⨂[𝕜] i, E i)) ⟨0, lifts_zero⟩ ?_
+      simp only [projectiveSeminormAux, Function.comp_apply]
+      rw [List.sum_eq_zero]
+      intro _
+      simp only [List.mem_map, Prod.exists, forall_exists_index, and_imp]
+      intro _ _ hxm
+      rw [← FreeAddMonoid.ofList_nil] at hxm
+      exfalso
+      exact List.not_mem_nil _ hxm
+    · letI : Nonempty (lifts 0) := ⟨0, lifts_zero (R := 𝕜) (s := E)⟩
+      exact le_ciInf (fun p ↦ projectiveSeminormAux_nonneg p.1)
+  · intro x y
+    letI := liftsNonempty x; letI := liftsNonempty y
+    exact le_ciInf_add_ciInf (fun p q ↦ ciInf_le_of_le (projectiveSemiNormAuxBddBelow _)
+      ⟨p.1 + q.1, lifts_add p q⟩ (projectiveSeminormAux_add_le p.1 q.1))
+  · intro a x
+    letI := liftsNonempty x
+    rw [Real.mul_iInf_of_nonneg (norm_nonneg _)]
+    refine le_ciInf ?_
+    intro p
+    rw [← projectiveSeminormAux_smul]
+    exact ciInf_le_of_le (projectiveSemiNormAuxBddBelow _) ⟨(List.map (fun y ↦ (a * y.1, y.2)) p.1),
+    lifts_smul p.2 a⟩ (le_refl _)
+
+theorem projectiveSeminorm_apply (x : ⨂[𝕜] i, E i) :
+    projectiveSeminorm x = iInf (fun (p : lifts x) ↦ projectiveSeminormAux p.1) := by rfl
+
+theorem projectiveSeminorm_tprod_le (m : Π i, E i) :
+    projectiveSeminorm (⨂ₜ[𝕜] i, m i) ≤ ∏ i, ‖m i‖ := by
+  rw [projectiveSeminorm_apply]
+  convert ciInf_le (projectiveSemiNormAuxBddBelow _) ⟨[((1 : 𝕜), m)] ,?_⟩
+  · simp only [projectiveSeminormAux, Function.comp_apply, List.map_cons, norm_one, one_mul,
+    List.map_nil, List.sum_cons, List.sum_nil, add_zero]
+  · rw [mem_lifts_iff, List.map_singleton, List.sum_singleton, one_smul]
+
+theorem projectiveSeminorm_bound (x : ⨂[𝕜] i, E i) (G : Type*) [SeminormedAddCommGroup G]
```

Comment:
Can you find a better name for this lemma? Here it's impossible to guess that one is applying a continuous multilinear map...

Reply:
What about `norm_apply_le_projectiveSeminorm` ? I can also make it `norm_continuousMultilinearMap_apply_le_projectiveSeminorm`, if it's not too long.

Reply:
Actually I'd rather make it `norm_eval_le_projectiveSeminorm` (and similarly `norm_eval_le_injectiveSeminorm` in `InjectiveSeminorm.lean`), since what I'm really bounding here is the norm of the evaluation map at `x`.